app/com/arpnetworking/rollups/MetricsDiscovery.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2019 Dropbox Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.rollups;
+
+import akka.actor.AbstractActorWithTimers;
+import akka.actor.ActorRef;
+import akka.actor.Status;
+import akka.pattern.PatternsCS;
+import com.arpnetworking.kairos.client.KairosDbClient;
+import com.arpnetworking.kairos.client.models.KairosMetricNamesQueryResponse;
+import com.arpnetworking.play.configuration.ConfigurationHelper;
+import com.arpnetworking.steno.Logger;
+import com.arpnetworking.steno.LoggerFactory;
+import com.typesafe.config.Config;
+import scala.concurrent.duration.FiniteDuration;
+
+import java.util.Iterator;
+import java.util.LinkedHashSet;
+import java.util.Optional;
+import java.util.Set;
+import java.util.regex.Pattern;
+import javax.inject.Inject;
+
+/**
+ * Actor for discovering the list of metrics available to be rolled up on a periodic basis.
+ *
+ * This actor maintains an internal set of metric names and acts as a source for downstream
+ * actors that perform the actual rollups.  This is intended to be used as a singleton in the
+ * cluster.
+ *
+ * @author Gilligan Markham (gmarkham at dropbox dot com)
+ */
+public class MetricsDiscovery extends AbstractActorWithTimers {
+
+    @Override
+    public Receive createReceive() {
+        return receiveBuilder()
+                .matchEquals(FETCH_MSG,
```

Comment:
Convention is wrap-all; e.g.

not:
```
foo(argA,
        argB,
        argC)
```

instead:
```
foo(
        argA,
        argB,
        argC)
```

Reply:
done